lib/generator.js

```diff
@@ -297,22 +306,22 @@ class Generator {
 
       walker.on('file', async (root, stats, next) => {
         try {
-          if (stats.name.includes('$$channel$$')) {
-            // this file should be handled for each in asyncapi.channels
-            await this.generateChannelFiles(asyncapiDocument, stats.name, root);
-            const templateFilePath = path.relative(this.templateDir, path.resolve(root, stats.name));
-            fs.unlink(path.resolve(this.targetDir, templateFilePath), next);
-          } else if (stats.name.includes('$$message$$')) {
-            // this file should be handled for each in asyncapi.components.messages
-            await this.generateMessageFiles(asyncapiDocument, stats.name, root);
-            const templateFilePath = path.relative(this.templateDir, path.resolve(root, stats.name));
-            fs.unlink(path.resolve(this.targetDir, templateFilePath), next);
-          } else if (stats.name.includes('$$securityScheme$$')) {
-            // this file should be handled for each in asyncapi.components.securityScheme
-            await this.generateSecuritySchemeFiles(asyncapiDocument, stats.name, root);
-            const templateFilePath = path.relative(this.templateDir, path.resolve(root, stats.name));
-            fs.unlink(path.resolve(this.targetDir, templateFilePath), next);
-          } else {
+          // Check if the filename dictates it should be separated
+          let wasSeparated = false;
+          for (const prop in fileNamesForSeperation) {
+            if (Object.prototype.hasOwnProperty.call(fileNamesForSeperation, prop)) {
+              if (stats.name.includes(`$$${prop}$$`)) {
+                await this.generateSeperateFiles(asyncapiDocument, fileNamesForSeperation[prop], prop, stats.name, root);
+                const templateFilePath = path.relative(this.templateDir, path.resolve(root, stats.name));
+                fs.unlink(path.resolve(this.targetDir, templateFilePath), next);
+                wasSeparated = true;
+                //The filename can only contain 1 specifier (message, scheme etc)
+                break;
+              }
+            }
+          }
+          // If it was not separated process normally
+          if (!wasSeparated) {
```

Comment:
Minor comment for the block above and some changes below too: you're using "seperation" and "separation" alternatively :)

Reply:
Yep, spelling mistake 😄